WordPress/Classes/ViewRelated/NewGutenberg/NewGutenbergViewController.swift

```diff
@@ -132,9 +132,37 @@ class NewGutenbergViewController: UIViewController, PostEditor, PublishingEditor
         self.navigationBarManager = navigationBarManager ?? PostEditorNavigationBarManager()
 
         let networkClient = NewGutenbergNetworkClient(blog: post.blog)
+
+        let selfHostedApiUrl = post.blog.url(withPath: "wp-json/")
+        let siteApiRoot = post.blog.isAccessibleThroughWPCom() ? post.blog.wordPressComRestApi()?.baseURL.absoluteString : selfHostedApiUrl
+        let siteId = post.blog.dotComID?.stringValue
+        let authToken = post.blog.authToken ?? ""
+        var authHeader = "Bearer \(authToken)"
+
+        let applicationPassword = try? post.blog.getApplicationToken()
+
+        if let appPassword = applicationPassword, let username = post.blog.username {
```

Comment:
Hey @crazytonyli 👋 since you've been working on the implementation to support application passwords. Do you think this is the best way to get this data? Any feedback on this or the PR in general would be appreciated, thank you!